contracts/provider/assembly/index.ts

```diff
@@ -0,0 +1,216 @@
+import {
+  u128,
+  storage,
+  context,
+  ContractPromiseBatch,
+  PersistentUnorderedMap,
+  logging,
+} from "near-sdk-as";
+import { onlyOwner } from "../../ownable/assembly";
+export {
+  transferOwnership,
+  renounceOwnership,
+  owner,
+} from "../../ownable/assembly";
+
+export const DefaultDivisor = u128.from("416000000000"); // yocto-Near per nano-second
+
+export const deposits = new PersistentUnorderedMap<string, Deposit>("d");
+
+@nearBindgen
+export class Deposit {
+  constructor(
+    public timestamp: u64,
+    public depositor: string,
+    public value: u128
+  ) {}
+}
+
+/**
+ * Proportion of deposited funds that will be kept by this provider upon release.
+ */
+export function providerProportion(): f64 {
```

Comment:
Since near doesn't have simple support for single-value (you have to interact with storage directly), we need all of these extra functions and private/hidden methods because we also want the public methods to be onlyOwner methods on the setters.